src/factories/workflow-specific/Restricted_PIM_Factory_v1.sol

```diff
@@ -0,0 +1,238 @@
+        _manageInitialSupplies(
+            IBondingCurveBase_v1(fundingManager),
+            IERC20(collateralToken),
+            issuanceToken,
+            bcProperties.initialCollateralSupply,
+            bcProperties.initialIssuanceSupply,
+            realAdmin,
+            _msgSender()
+        );
+
+        // assign permissions to buy/sell from curve to admin
+        bytes32 curveAccess = FM_BC_Restricted_Bancor_Redeeming_VirtualSupply_v1(
+            fundingManager
+        ).CURVE_INTERACTION_ROLE();
+        FM_BC_Restricted_Bancor_Redeeming_VirtualSupply_v1(fundingManager)
+            .grantModuleRole(curveAccess, _msgSender());
+        // revoke minting rights from factory
+        issuanceToken.setMinter(address(this), false);
+
+        // revoke privileges from factory
+        _transferTokenOwnership(issuanceToken, realAdmin);
+        _transferWorkflowAdminRights(orchestrator, realAdmin);
+
+        emit IRestricted_PIM_Factory_v1.PIMWorkflowCreated(
+            address(orchestrator), address(issuanceToken), _msgSender()
+        );
+    }
+
+    /// @inheritdoc IRestricted_PIM_Factory_v1
+    function addFunding(address actor, address token, uint amount) external {
+        // records funding amount
+        fundings[_msgSender()][actor][token] += amount;
+        // sends amount from msg.sender to factory
+        IERC20(token).safeTransferFrom(_msgSender(), address(this), amount);
+
+        emit IRestricted_PIM_Factory_v1.FundingAdded(
+            _msgSender(), actor, token, amount
+        );
+    }
+
+    /// @inheritdoc IRestricted_PIM_Factory_v1
+    function withdrawFunding(address actor, address token, uint amount)
+        external
+    {
+        // checks if the requested amount is available
+        uint availableFunding = fundings[_msgSender()][actor][token];
+        if (amount > availableFunding) {
+            revert IRestricted_PIM_Factory_v1.InsufficientFunding(
+                availableFunding
+            );
+        }
+        // if so adjusts internal balancing
+        fundings[_msgSender()][actor][token] -= amount;
+        // and sends amount to msg sender
+        IERC20(token).safeTransfer(_msgSender(), amount);
+
+        emit IRestricted_PIM_Factory_v1.FundingRemoved(
+            _msgSender(), actor, token, amount
+        );
+    }
+
+    //--------------------------------------------------------------------------
+    // Internal Functions
+
+    function _manageInitialSupplies(
+        IBondingCurveBase_v1 fundingManager,
+        IERC20 collateralToken,
+        ERC20Issuance_v1 issuanceToken,
+        uint initialCollateralSupply,
+        uint initialIssuanceSupply,
+        address admin,
+        address actor
+    ) private {
+        uint availableFunding = fundings[admin][actor][address(collateralToken)];
+        if (availableFunding < initialCollateralSupply) {
```

Comment:
As it reverts here this basically means that it's not an option to pay this as the deployer in that specific moment, there needs to be a sponsor that pre-funded, right? Yes, the actor can be the sponsor, but this entails two calls (first `addFunding` before creating the workflow. Is that intentional? In my mind a fallback mechanism would make sense where the sponsors funds are taken, and if not enough, the deployer is paying (if the approved it)? But if that's not required for the client or I am just missing something then feel free to ignore this ofc.

Reply:
In this case the client will pre-seed the contract with the funds, since projects aren't supposed to pay anything. A fallback mechanism makes sense, but is somewhat out of scope for this case?

Reply:
If you don't mind I'll keep it qacc-specific and refrain from adding the fallback.